### PR TITLE
refactor: add sort by name and query performance

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Adds sort by name again and refactors the query for performance, to test ensure that sorting still works fine and all places where we use collections still works (create gallery etc)

Closes https://app.clickup.com/t/862kehzwv

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Review by samharperpittam — CHANGES_REQUESTED** [review]
When changing the view from table to grid view, Collections do not appear to be sorted by Name. 

Though the performance is super fast in table view now.

File: app/Http/Controllers/CollectionController.php [hunk 1]
```diff
@@ -87,9 +87,9 @@ public function index(Request $request): Response|JsonResponse|RedirectResponse
                 ->when($showHidden, fn ($q) => $q->whereIn('collections.id', $user->hiddenCollections->modelKeys()))
                 ->when(! $showHidden, fn ($q) => $q->whereNotIn('collections.id', $user->hiddenCollections->modelKeys()))
                 ->when(count($selectedChainIds) > 0, fn ($q) => $q->whereIn('collections.network_id', Network::whereIn('chain_id', $selectedChainIds)->pluck('id')))
-                ->when($sortBy === 'name', fn ($q) => $q->orderBy('name', $sortDirection))
+                ->when($sortBy === 'name' || $sortBy === null, fn ($q) => $q->orderByName($sortDirection))
                 ->when($sortBy === 'floor-price', fn ($q) => $q->orderByFloorPrice($sortDirection, $user->currency()))
-                ->when($sortBy === 'value' || $sortBy === null, fn ($q) => $q->orderByValue($user->wallet, $sortDirection, $user->currency()))
```
Comment (ItsANameToo) [inline app/Http/Controllers/CollectionController.php:92]: make value the default again, as `name` was meant as a temporary fix when the floor prices were missing

**Review by ItsANameToo — APPROVED** [review]

**Comment by ItsANameToo** [comment]
> When changing the view from table to grid view, Collections do not appear to be sorted by Name.
> 
> Though the performance is super fast in table view now.

https://app.clickup.com/t/862kfhg2v

**Comment by ItsANameToo** [comment]
@alfonsobries can you rebase this onto `rc-0.7.0` so it can be included in that release

**Comment by alfonsobries** [comment]
@ItsANameToo The base branch is now `rc-0.7.0`. I made a mess when doing the rebase, including some force-pushes. I mention it because you might need to update the branch locally and, secondly, to make sure I didn't delete any of the changes you made

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**